### PR TITLE
server connected to mongoDB, tested with Postman

### PR DESCRIPTION
**Pull request by borisolah**
Fixes issue #18 Setting up Express and MongoDB as the backend technologies.

Successfully set up the MongoDB Atlas and connected it to the Express server, tested login and register functions via  Postman, hashed and salted the passwords using bycript and handled a few error messages such as password too short, invalid E-mail, or user already registered.

### CODE REVIEW

File: server/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Object.values(err.errors).forEach(({ properties }) => {
+      errors[properties.path] = properties.message;
+    });
+  }
+  return errors;
+};
+
+const maxAge = 3 * 24 * 60 * 60;
+const createToken = (id) => {
+  return jwt.sign({ id }, "string secret comes here", {
```
Comment (csillagkrisztian) [inline server/controllers/authController.js:25]: What does that "string secret comes here" mean? If you are supposed to be generating a secret, is that hard to do?

File: server/controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    expiresIn: maxAge,
```
Comment (csillagkrisztian) [inline server/controllers/authController.js:26]: Also, what is maxAge? JWT tokens should expire in a reasonable amount of time, let's say a week for security's sake, so people don't forget their credentials.

File: server/controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  });
+};
+
+// controller actions
+module.exports.signup_get = (req, res) => {
+  res.render("signup");
```
Comment (csillagkrisztian) [inline server/controllers/authController.js:32]: are you rendering stuff on the server?

File: server/controllers/authController.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  res.send("login");
+};
+
+module.exports.signup_post = async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.create({ email, password });
+    const token = createToken(user._id);
+    res.cookie("jwt", token, { httpOnly: true, maxAge: maxAge * 1000 });
```
Comment (csillagkrisztian) [inline server/controllers/authController.js:45]: Why is it maxAge *1000?

File: server/server.js [hunk 5]
```diff
@@ -1,34 +1,28 @@
...
-app.get("/", (req, res) => {
-  res.sendFile(__dirname + "/views/index.html");
-});
+app.use(cookieParser());
+app.use(cors());
 
-app.get("/", (req, res) => {
-  res.sendFile(__dirname + "/views/index.html");
-});
+// view engine
```
Comment (csillagkrisztian) [inline server/server.js:15]: What do you mean by engine?

**Review by csillagkrisztian — CHANGES_REQUESTED** [review]
Super clean PR! Looks professional! Although I have a few questions and ideally you would add a comment before big functions to explain in short what they are doing and why!

File: server/controllers/authController.js (hunk 4 above)
Comment (borisolah) [reply server/controllers/authController.js:45]: So this part of the code required miliseconds, and not seconds, so I just multiplied the whole 3 days by 1000 and that should have given me 3 days in miliseconds.

File: server/controllers/authController.js (hunk 1 above)
Comment (borisolah) [reply server/controllers/authController.js:25]: This and the mongoDB Atlas usernape/password part are a bit difficult to implement. I'm going to have to use specific keys to gain access to the databse from the server, and I did not want to upload those tot he github hence had a string telling me where to put it in the future.

I will need to do excessive research on how to do this properly in a more professional manner.

File: server/controllers/authController.js (hunk 3 above)
Comment (borisolah) [reply server/controllers/authController.js]: Thank you for pointing this out, somwhow I've missed it. I wanted to test something out with singing up without a React client and forgot to change it back after. 

Fixed it up just now.

File: server/controllers/authController.js (hunk 1 above)
Comment (csillagkrisztian) [reply server/controllers/authController.js:25]: To use secret strings you should use environment variables
They are .env files, that should not be pushed to the repo, but can be local on your machine.

File: server/controllers/authController.js (hunk 2 above)
Comment (csillagkrisztian) [reply server/controllers/authController.js:26]: sure, but then maxAge refers to the age of the token not the user, I think this is confusing, so I would prefer that you name it, expirationDate or Time

Also if you want to make it even more clear, just do it this way
const dayInSeconds = 86400;
const expiryDate = 3 * dayInSeconds;
This way it's clear what the day is and what the rest is as well.

Comment (borisolah) [reply server/controllers/authController.js:26]: That is true. I will fix it up and make a new commit in a minute.

File: server/controllers/authController.js (hunk 1 above)
Comment (csillagkrisztian) [reply server/controllers/authController.js:27]: I have to comment again! Either decide to use shorthands or never use them ever. Usually the never use them ever is the winner. It makes the code a bit large but the readabolity is massive

**Review by csillagkrisztian — CHANGES_REQUESTED** [review]

File: server/controllers/authController.js (hunk 1 above)
Comment (borisolah) [reply server/controllers/authController.js:27]: All right, I'll keep it with not using shorthands then.

**Review by csillagkrisztian — APPROVED** [review]

**Pull request merged**